utils/mountinfo.c

```diff
@@ -0,0 +1,83 @@
+#include <stdio.h>
+#include <string.h>
+
+#include "utils/mountinfo.h"
+#include "utils/utils.h"
+
+bool find_tracing_dir(char **tracing_dir)
+{
+	FILE *fp;
+	char *line = NULL, *ptr, *end, mnt[NAME_MAX];
+	size_t len;
+	int i;
+
+	fp = fopen(PROC_MOUNTINFO, "r");
+	if (fp == NULL)
+		return false;
+
+	/* First, try to find tracefs. */
+	while (getline(&line, &len, fp) > 0) {
+		if (!(ptr = strstr(line, "tracefs")))
+			continue;
+
+		/*
+                 * /proc/self/mountinfo format:
+                 * 36 35 98:0 /mnt1 /mnt2 rw,noatime master:1 - ext3
+                 * (1)(2)(3)   (4)   (5)      (6)      (7)   (8) (9)
+                 *
+                 * Skip 4 spaces to retrieve (5) mount point.
+                 */
+		ptr = line;
+		for (i = 0; i < 4; i++) {
+			ptr = strchr(ptr, ' ');
+			if (ptr == NULL)
+				break;
+			ptr++;
+		}
+		if (ptr == NULL)
+			continue;
+
+		end = strchr(ptr, ' ');
+		if (end - ptr >= NAME_MAX / 2) /* FIXME? */
+			continue;
+		strncpy(mnt, ptr, (size_t)(end - ptr));
+		xasprintf(tracing_dir, "%s", mnt);
+		pr_dbg2("tracefs found at %s\n", mnt);
+		pr_dbg2("TRACING_DIR: %s\n", *tracing_dir);
+		return true;
+	}
+
+	/*
+         * If tracefs is not found (e.g., older kernel), try to find debugfs
+         * and use its subdirectory `tracing` as `TRACING_DIR`.
+         */
```

Comment:
This doesn't follow the coding style.  Please do not indent the comment like this.